src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -222,3 +224,94 @@ export const archiveForm = (
     // On success, return true
   }).map(() => true)
 }
+
+/**
+ * Transfer form ownership from current owner to the new email.
+ * @param currentForm the form to transfer ownership for
+ * @param newOwnerEmail the email of the new owner to transfer to
+ *
+ * @return ok(updated form) if transfer is successful
+ * @return err(MissingUserError) if the current form admin cannot be found
+ * @return err(TransferOwnershipError) if new owner cannot be found in the database or new owner email is same as current owner
+ * @return err(DatabaseError) if any database errors like missing admin of current owner occurs
+ */
+export const transferFormOwnership = (
+  currentForm: IPopulatedForm,
+  newOwnerEmail: string,
+): ResultAsync<IPopulatedForm, TransferOwnershipError | DatabaseError> => {
+  const logMeta = {
+    action: 'transferFormOwnership',
+    newOwnerEmail,
+  }
+
+  return (
+    // Step 1: Retrieve current owner of form to transfer.
+    findAdminById(currentForm.admin._id)
+      .andThen((currentOwner) =>
+        // Step 2: Retrieve user document for new owner.
+        findAdminByEmail(newOwnerEmail)
+          .mapErr((error) => {
+            logger.error({
+              message:
+                'Error occurred whilst finding new owner email to transfer ownership to',
+              meta: logMeta,
+              error,
+            })
+
+            // Override MissingUserError with more specific message if new owner
+            // cannot be found.
+            if (error instanceof MissingUserError) {
+              return new TransferOwnershipError(
+                `${newOwnerEmail} must have logged in once before being added as Owner`,
+              )
+            }
+            return error
+          })
+          // Step 3: Perform form ownership transfer.
+          .andThen((newOwner) => {
+            // No need to transfer form ownership if new and current owners are
+            // the same.
+            if (newOwner.email === currentOwner.email) {
```

Comment:
this check can be done earlier right? to remove one unnecessary network call

Reply:
nice catch!!!